db/write_batch.cc

```diff
@@ -2756,16 +2756,10 @@ class ProtectionInfoUpdater : public WriteBatch::Handler {
 
 Status WriteBatchInternal::SetContents(WriteBatch* b, const Slice& contents) {
   assert(contents.size() >= WriteBatchInternal::kHeader);
+  assert(b->prot_info_ == nullptr);
```

Comment:
This assert is meant more like "not supported yet" , not a fundamental part of the API. It's not supported yet because it's unclear as of now how we want to have checksum coverage over `contents` through `b->rep_` and I felt the old support for updating `prot_info_` in `SetContents()` could be misleading. If it's a public API we should definitely return `Status::NotSupported`. But it's internal/uncommonly used so I was guessing this is fine...